src/event/quic/ngx_event_quic_protection.c

```diff
@@ -122,9 +122,10 @@ ngx_quic_keys_set_initial_secret(ngx_quic_keys_t *keys, ngx_str_t *secret,
     ngx_quic_secret_t   *client, *server;
     ngx_quic_ciphers_t   ciphers;
 
-    static const uint8_t salt[20] =
-        "\x38\x76\x2c\xf7\xf5\x59\x34\xb3\x4d\x17"
-        "\x9a\xe6\xa4\xc8\x0c\xad\xcc\xbb\x7f\x0a";
+    static const uint8_t salt[20] = {
+        0x38, 0x76, 0x2c, 0xf7, 0xf5, 0x59, 0x34, 0xb3, 0x4d, 0x17,
+        0x9a, 0xe6, 0xa4, 0xc8, 0x0c, 0xad, 0xcc, 0xbb, 0x7f, 0x0a,
+    };
```

Comment:
I don't think this is a change in the right direction.

What I'd like to see is declaring with unspecified array size, e.g. similar to the grpc module. This allows to keep using the initialization string unmodified, written in a more compact form, compared to a byte sequence.

Calculating the number of array elements with `sizeof() - 1` is a common approach used in nginx as well.

```
$ grep -E 'sizeof\([^)]+\) - 1' src/http/v2/ngx_http_v2_filter_module.c
    (ngx_http_v2_integer_octets(sizeof(h) - 1) + sizeof(h) - 1)
            len += 1 + sizeof(nginx) - 1;
            len += sizeof("; charset=") - 1 + r->headers_out.charset.len;
            location.len = sizeof("https://") - 1 + host.len
                location.len += sizeof(":65535") - 1;
            p = ngx_cpymem(location.data, "http", sizeof("http") - 1);
            len += 1 + sizeof(accept_encoding) - 1;
                                            sizeof(NGINX_VER) - 1, tmp);
                                            sizeof(NGINX_VER_BUILD) - 1, tmp);
            pos = ngx_cpymem(pos, nginx, sizeof(nginx) - 1);
            len = r->headers_out.content_type.len + sizeof("; charset=") - 1
            p = ngx_cpymem(p, "; charset=", sizeof("; charset=") - 1);
        len = sizeof("Wed, 31 Dec 1986 18:00:00 GMT") - 1;
        pos = ngx_cpymem(pos, accept_encoding, sizeof(accept_encoding) - 1);
```
 This covers both C arrays and literal strings.

As a bonus, this allows logging the `salt` value without variable length modifier:

```
@@ -152,8 +152,8 @@ ngx_quic_keys_set_initial_secret(ngx_quic_keys_t *keys, ngx_str_t *secret,
     ngx_log_debug0(NGX_LOG_DEBUG_EVENT, log, 0,
                    "quic ngx_quic_set_initial_secret");
 #ifdef NGX_QUIC_DEBUG_CRYPTO
-    ngx_log_debug3(NGX_LOG_DEBUG_EVENT, log, 0,
-                   "quic salt len:%uz %*xs", sizeof(salt), sizeof(salt), salt);
+    ngx_log_debug2(NGX_LOG_DEBUG_EVENT, log, 0,
+                   "quic salt len:%uz %xs", sizeof(salt) - 1, salt);
     ngx_log_debug3(NGX_LOG_DEBUG_EVENT, log, 0,
                    "quic initial secret len:%uz %*xs", is_len, is_len, is);
 #endif
```